packages/react-components/react-table/src/hooks/useColumnSizing.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  TableColumnId,
+  ColumnWidthProps,
+  ColumnWidthState,
+  TableColumnSizingState,
+  TableFeaturesState,
+  UseColumnSizingParams,
+} from './types';
+import { useColumnResizeState } from './useColumnResizeState';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import useColumnResizeMouseHandler from './useColumnResizeMouseHandler';
+import { useMeasureElement } from './useMeasureElement';
+
+export const defaultColumnSizingState: TableColumnSizingState = {
+  getColumnWidth: () => 0,
+  getColumnWidths: () => [],
+  getOnMouseDown: () => () => null,
+  getTotalWidth: () => 0,
+  setColumnWidth: () => null,
+  getColumnProps: () => ({ style: {}, columnId: '' }),
+};
+
+export function useColumnSizing_unstable<TItem>(params?: UseColumnSizingParams) {
+  // False positive, these plugin hooks are intended to be run on every render
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  return (tableState: TableFeaturesState<TItem>) => useColumnSizingState(tableState, params);
+}
+
+function getColumnProps(column: ColumnWidthState): ColumnWidthProps {
+  const width = column.width;
+  return {
+    columnId: column.columnId,
+    style: {
+      // native styles
+      width,
+      // non-native element styles (flex layout)
+      minWidth: width,
+      maxWidth: width,
+    },
+  };
+}
+
+function useColumnSizingState<TItem>(
+  tableState: TableFeaturesState<TItem>,
+  params?: UseColumnSizingParams,
+): TableFeaturesState<TItem> {
+  const { columns, tableRef } = tableState;
+
+  const { targetDocument } = useFluent();
+  const win = targetDocument?.defaultView;
+
+  // Gets the container width
+  const containerWidth = useMeasureElement(tableRef);
+  // Creates the state based on columns and available containerWidth
+  const columnResizeState = useColumnResizeState<TItem>(columns, containerWidth, params);
+  // Creates the mouse handler and attaches the state to it
+  const mouseHandler = useColumnResizeMouseHandler(columnResizeState, win);
```

Comment:
```suggestion
  const mouseHandler = useColumnResizeMouseHandler(columnResizeState);
```

Please consume `targetDocument` directly in the `useColumnResizeMouseHandler()`.